src/impl_windows.rs

```diff
@@ -43,11 +39,49 @@ pub fn runas_impl(cmd: &Command) -> io::Result<ExitStatus> {
         .collect::<Vec<_>>();
 
     unsafe {
-        let show = if cmd.hide { 0 } else { 1 };
-        Ok(mem::transmute(rust_win_runas(
-            file.as_ptr(),
-            params.as_ptr(),
-            show,
-        )))
+        use winapi::um::combaseapi::CoInitializeEx;
+        use winapi::um::objbase::{COINIT_APARTMENTTHREADED, COINIT_DISABLE_OLE1DDE};
+        use winapi::um::shellapi::{SHELLEXECUTEINFOW, ShellExecuteExW, SEE_MASK_NOASYNC, SEE_MASK_NOCLOSEPROCESS};
+        use winapi::um::winbase::INFINITE;
+        use winapi::um::winuser::{SW_HIDE, SW_NORMAL};
+        use winapi::shared::minwindef::{DWORD, FALSE};
+        use winapi::um::synchapi::WaitForSingleObject;
+        use winapi::um::processthreadsapi::GetExitCodeProcess;
+        use std::ptr;
+
+        let show = if cmd.hide { SW_HIDE } else { SW_NORMAL };
+
+        let mut code : DWORD = 0;
+        let mut sei = SHELLEXECUTEINFOW { 
+            cbSize: mem::size_of::<SHELLEXECUTEINFOW>() as u32,
+            fMask: SEE_MASK_NOASYNC | SEE_MASK_NOCLOSEPROCESS,
+            lpVerb: (*"runas").encode_utf16().collect::<Vec<u16>>().as_ptr(),
+            lpFile: file.as_ptr(),
+            lpParameters: params.as_ptr(),
+            nShow: show,
+            dwHotKey: 0,
+            hInstApp: ptr::null_mut(),
+            hMonitor: ptr::null_mut(),
+            hProcess: ptr::null_mut(),
+            hkeyClass: ptr::null_mut(),
+            hwnd: ptr::null_mut(),
+            lpClass: ptr::null_mut(),
+            lpDirectory: ptr::null_mut(),
+            lpIDList: ptr::null_mut(),
+        };
+
+        CoInitializeEx(ptr::null_mut(), COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
```

Comment:
Technically, there should be a matching call to `CoUninitialize()` for each call to `CoInitializeEx()`. [Here](https://github.com/amodm/webbrowser-rs/blob/1ce016d3eccdf974be89c415e7afc5b225c96fe2/src/windows.rs#L26) is an example of it done in another crate.

Reply:
Adapted it so that each call is met with an `CoUnInitialize`